app/web/components/Map.tsx

```diff
@@ -84,9 +84,6 @@ export default function Map({
   useEffect(() => {
     if (!containerRef.current) return;
 
-    //don't create a new map if it exists already
-    if (mapRef.current) return;
-
```

Comment:
Might this be related to the flickering?

Reply:
So this one nearly killed me. I literally woke up this morning thinking about this FROM A DEAD SLEEP lol.

The culprit: In React 18, if you have Strict Mode enabled, the useEffect hook runs twice in development mode. This was causing this to return before the map was initialized so it was blank. So I am storing an `isMapInitialized` value in state now instead and referencing it and that seems to do the trick.

Reply:
I think this type of `useEffect`s is exactly what the Strict Mode change is trying to catch - asymmetrical effects that need clean ups!

Would moving the separate cleanup below on line 126 into this effect instead negate the need to use the `isMapInitialized` trick? There's actually a warning in the official docs that [advise against doing this to prevent the effect from firing twice](https://react.dev/learn/synchronizing-with-effects#how-to-handle-the-effect-firing-twice-in-development) if possible so we're not hiding subtle bugs.

Reply:
@darrenvong Not the React docs calling me out...lol that they literally have a map example in the description. Thank you so much this really broke my brain. Last time I was a frontend engineer in my day job we were using React v17 hah.

I did almost this with a few adjustments. I needed to make the useEffect only load on first render otherwise it rerenders whenever I type in the location input. And needed to set `mapRed.current = undefined` in addition t removing the mapRef.

I checked the MapSearch map and that's still working too so 🤞.

Reply:
Right, yeah there's some nasty stuff to handle if we were to follow the exhaustive deps rule which might be tricky to deal with without making some values more stable, and that'd probably complicate the code more. Without knowing too much about this whole thing works (the Map is the area of app I didn't really work with as closely 😅), this seems like a reasonable tradeoff until we refactor further I reckon 👍

Reply:
Or if you somehow know which variables may make sense to still be reactive over from more recent working knowledge of this beast, I'd recommend following this part of the doc to pull out the non-reactive variables into an "effect event" - https://react.dev/learn/separating-events-from-effects#choosing-between-event-handlers-and-effects

Although this whole technique is still a bit clunky imo, since the `useEffectEvent` hook doesn't actually exist as an official API, so you'd have to add a [rough userland implementation of it](https://gist.github.com/AndrewIngram/2e5d530eb91b96c6a1b009b21c59db52), I'll leave it up to you on this one on whether you want to go that far to follow the official advice!

Reply:
So I actually think everything in that useEffect is for initial load, so it makes sense to be in a useEffect. The onClick, zoom, etc. etc. are in separate functions.

I'm also a bit overwhelmed by the map TBH. I think it could use a refactor to make it more readable, but since we're likely going to do that as part of a map redesign soon-ish it doesn't seem like a priority.

What makes it complicated IMO is that we're using this Map component multiple places, but we've put in all this conditional logic depending on where it's being rendered.

Probably this gets way cleaner if we opt to duplicate *some* logic and just break them out into separate map components by use case. Then we don't need different map wrapper components either I assume, which is also causing rerenders for example when typing in the location input even before we hit enter.

So for now just trying to get it to work in a decent, not-too-hacky way until we decide on the map design then I will go at it with a 🔪 if I have anything to say about it.